aiohttp/formdata.py

```diff
@@ -115,6 +121,8 @@ def _gen_form_urlencoded(self) -> payload.BytesPayload:
 
     def _gen_form_data(self) -> multipart.MultipartWriter:
         """Encode a list of fields using the multipart/form-data MIME format"""
+        if self._is_processed:
+            raise ClientPayloadError('Form data has been processed already')
```

Comment:
This is the programming error that don't depend on networking etc.
The error should be fixed by changing the source code, `raise RuntimeError` is the right exception in this context.